### ISSUE TITLE
Remote terminal hangs at "Starting..."

### ISSUE THREAD

**Issue by rob-mccutcheon** [microsoft/vscode-remote-release#3320]
- VSCode Version:1.46.1
- Local OS Version: OSX Mojave 10.14.4
- Remote OS Version: Scientific Linux 7.8 (Nitrogen)
- Remote Extension/Connection Type: SSH/

Steps to Reproduce:

1. Start a new remote server - I am able to access and edit files
2. Remote terminal does not load, no error message but hangs with "Starting..." in the dropdown

This seems similar to -
https://github.com/microsoft/vscode-remote-release/issues/1154
But I do not receive any error message, and there seems to be nothing of relevance in the develop log files:

```

[2020-07-06 08:42:57.967] [remoteagent] [info] Extension host agent started.
[2020-07-06 08:42:58.381] [remoteagent] [info] [127.0.0.1][e9459a3e][ManagementConnection] New connection established.
[2020-07-06 08:42:58.386] [remoteagent] [info] [127.0.0.1][82f9fe50][ExtensionHostConnection] New connection established.
[2020-07-06 08:42:58.404] [remoteagent] [info] [127.0.0.1][82f9fe50][ExtensionHostConnection] <117699> Launched Extension Host Process.
[2020-07-06 08:43:07.580] [remoteagent] [info] [127.0.0.1][e9459a3e][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2020-07-06 08:44:19.658] [remoteagent] [info] [127.0.0.1][49fe4e9e][ManagementConnection] New connection established.
[2020-07-06 08:44:19.661] [remoteagent] [info] [127.0.0.1][f8442796][ExtensionHostConnection] New connection established.
[2020-07-06 08:44:19.667] [remoteagent] [info] [127.0.0.1][f8442796][ExtensionHostConnection] <118110> Launched Extension Host Process.
[2020-07-06 08:44:34.967] [remoteagent] [info] [127.0.0.1][49fe4e9e][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2020-07-06 08:45:13.917] [remoteagent] [info] [127.0.0.1][81ef159a][ManagementConnection] New connection established.
[2020-07-06 08:45:13.920] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] New connection established.
[2020-07-06 08:45:13.924] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] <118362> Launched Extension Host Process.
[2020-07-06 08:45:39.005] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:46:04.087] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:46:29.162] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:46:54.256] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:47:19.340] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:47:44.412] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:48:09.484] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:48:34.567] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:48:59.646] [remoteagent] [info] [127.0.0.1][d03fd239][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:49:17.457] [remoteagent] [info] [127.0.0.1][81ef159a][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2020-07-06 08:51:43.135] [remoteagent] [info] [127.0.0.1][51227c91][ManagementConnection] New connection established.
[2020-07-06 08:51:43.135] [remoteagent] [info] [127.0.0.1][1b9e3256][ExtensionHostConnection] New connection established.
[2020-07-06 08:51:43.142] [remoteagent] [info] [127.0.0.1][1b9e3256][ExtensionHostConnection] <119703> Launched Extension Host Process.
[2020-07-06 08:52:00.080] [remoteagent] [info] [127.0.0.1][51227c91][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2020-07-06 08:52:31.759] [remoteagent] [info] [127.0.0.1][bc91575e][ManagementConnection] New connection established.
[2020-07-06 08:52:31.760] [remoteagent] [info] [127.0.0.1][bdff4005][ExtensionHostConnection] New connection established.
[2020-07-06 08:52:31.766] [remoteagent] [info] [127.0.0.1][bdff4005][ExtensionHostConnection] <119978> Launched Extension Host Process.
[2020-07-06 08:52:56.858] [remoteagent] [info] [127.0.0.1][bdff4005][ExtensionHostConnection] The client has reconnected.
[2020-07-06 08:53:09.843] [remoteagent] [info] [127.0.0.1][bc91575e][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2020-07-06 08:53:57.137] [remoteagent] [info] [127.0.0.1][23fb0c04][ManagementConnection] New connection established.
[2020-07-06 08:53:57.138] [remoteagent] [info] [127.0.0.1][ae3128d4][ExtensionHostConnection] New connection established.
[2020-07-06 08:53:57.144] [remoteagent] [info] [127.0.0.1][ae3128d4][ExtensionHostConnection] <120284> Launched Extension Host Process.
[2020-07-06 08:53:57.198] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:53:57.203] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:53:57.206] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:53:57.223] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:53:57.233] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:53:57.274] [remoteagent] [trace] {"errno":-2,"code":"ENOENT","syscall":"open","path":"/home/k1201869/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/extensions/node_modules/package.json"}
[2020-07-06 08:53:57.275] [remoteagent] [trace] {"errno":-2,"code":"ENOENT","syscall":"open","path":"/home/k1201869/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/extensions/node_modules/package.json"}
[2020-07-06 08:53:57.275] [remoteagent] [trace] {"errno":-2,"code":"ENOENT","syscall":"open","path":"/home/k1201869/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/extensions/node_modules/package.json"}
[2020-07-06 08:53:57.276] [remoteagent] [trace] {"errno":-2,"code":"ENOENT","syscall":"open","path":"/home/k1201869/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/extensions/node_modules/package.json"}
[2020-07-06 08:53:57.290] [remoteagent] [trace] Scanned system extensions: 30
[2020-07-06 08:53:57.303] [remoteagent] [trace] Scanned system extensions: 30
[2020-07-06 08:53:57.314] [remoteagent] [trace] Scanned system extensions: 30
[2020-07-06 08:53:57.323] [remoteagent] [trace] Scanned system extensions: 30
[2020-07-06 08:54:03.135] [remoteagent] [info] [127.0.0.1][23fb0c04][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2020-07-06 08:54:24.149] [remoteagent] [info] [127.0.0.1][f2e61827][ManagementConnection] New connection established.
[2020-07-06 08:54:24.150] [remoteagent] [trace] [127.0.0.1][a820df43][ExtensionHostConnection] - startParams language: en
[2020-07-06 08:54:24.150] [remoteagent] [trace] [127.0.0.1][a820df43][ExtensionHostConnection] - startParams env: undefined
[2020-07-06 08:54:24.150] [remoteagent] [info] [127.0.0.1][a820df43][ExtensionHostConnection] New connection established.
[2020-07-06 08:54:24.155] [remoteagent] [info] [127.0.0.1][a820df43][ExtensionHostConnection] <120497> Launched Extension Host Process.
[2020-07-06 08:54:24.185] [remoteagent] [trace] Scanning extensions using UI language: en
[2020-07-06 08:54:24.186] [remoteagent] [trace] Started scanning system extensions
[2020-07-06 08:54:24.186] [remoteagent] [trace] Started scanning user extensions
[2020-07-06 08:54:24.186] [remoteagent] [trace] Started scanning system extensions
[2020-07-06 08:54:24.186] [remoteagent] [trace] Started scanning user extensions
[2020-07-06 08:54:24.186] [remoteagent] [trace] Started scanning system extensions
[2020-07-06 08:54:24.186] [remoteagent] [trace] Started scanning user extensions
[2020-07-06 08:54:24.187] [remoteagent] [trace] Started scanning user extensions
[2020-07-06 08:54:24.187] [remoteagent] [trace] Started scanning system extensions
[2020-07-06 08:54:24.187] [remoteagent] [trace] Started scanning user extensions
[2020-07-06 08:54:24.187] [remoteagent] [trace] Started scanning system extensions
[2020-07-06 08:54:24.187] [remoteagent] [trace] Started scanning user extensions
[2020-07-06 08:54:24.223] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:54:24.228] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:54:24.238] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:54:24.258] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:54:24.264] [remoteagent] [trace] Scanned user extensions: 1
[2020-07-06 08:54:24.304] [remoteagent] [trace] Scanned user extensions: 1

```

Does this issue occur when you try this locally?:No

**Comment by sameronline** [comment]
Same here, no errors no nothing just stuck at starting, any quick workarounds?

**Comment by sameronline** [comment]
The issue is gone after i restarted the remote machine i'm trying to connect to (not sure why)

**Comment by oakad** [comment]
I'm seeing this as well after upgrading to 1.46.1

Both machines are Linux, the remote machine has 466 days uptime and was working with VSCode before.

I could not establish what may be wrong:

[17:05:37.637] Found local server running: {"agentPort":42143,"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/run/user/1292470249","platform":"linux","pid":8939,"ipcHandlePath":"/run/user/0/vscode-ssh-askpass-dc5f2d6b93da0bda3467ef65ce956c9c85c6c81e.sock","socksPort":42399,"startupTime":1594103853628}
[17:05:37.643] Found running server - short-circuiting install
[17:05:37.648] Starting forwarding server. localPort 45857 -> socksPort 42399 -> remotePort 42143
[17:05:37.648] Forwarding server listening on 45857
[17:05:37.648] Waiting for ssh tunnel to be ready
[17:05:37.677] [Forwarding server 45857] Got connection 0
[17:05:37.678] Tunneled remote port 42143 to local port 45857
[17:05:37.678] Resolved "ssh-remote+xxx" to "127.0.0.1:45857"
[17:05:37.685] ------




[17:05:37.703] [Forwarding server 45857] Got connection 1
[17:05:37.703] [Forwarding server 45857] Got connection 2
[17:06:05.750] [Forwarding server 45857] Got connection 3
[17:06:48.162] [Forwarding server 45857] Got connection 4
[17:07:36.485] [Forwarding server 45857] Got connection 5
[17:08:23.792] [Forwarding server 45857] Got connection 6
[17:09:12.747] [Forwarding server 45857] Got connection 7
[17:09:57.797] [Forwarding server 45857] Got connection 8
[17:10:45.522] [Forwarding server 45857] Got connection 9

**Comment by oakad** [comment]
It also appears that the remote extensions are not being started for me (probably, for the same reason that shell is not opening), despite the reassuring log message:

[2020-07-07 17:57:00.484] [exthost] [info] extension host started

**Comment by robmcc10** [comment]
I managed to get things working by connecting to another remote machine (that is identical to the one causing the problem)

**Comment by oakad** [comment]
More findings, this time on v1.47.2

If I wipe all the remote vscode state clean, vscode is able to initialize the remote side and opens some remote terminals successfully.

However, after I install some extensions, this stops working - the terminals won't open anymore and remote extensions are not started (no language servers are running).

Also, even without any extensions, ssh remoting feels sluggish compared to how it used to be couple of versions ago.

In short, there's clearly some issue with remote extension activation which projects onto terminals.

**Comment by roblourens** [comment]
https://github.com/microsoft/vscode-remote-release/issues/2313
